framework/src/kernels/EigenKernel.C

```diff
@@ -38,6 +38,9 @@ EigenKernel::EigenKernel(const InputParameters & parameters)
         dynamic_cast<MooseEigenSystem *>(&_fe_problem.getNonlinearSystemBase(_sys.number()))),
     _eigenvalue(NULL)
 {
+  if (!_eigen_sys)
+    mooseError("EigenKernel System is not a MooseEigenSystem.");
```

Comment:
```suggestion
    mooseError("Eigen kernel ", name(), " requires a MooseEigenSystem and was designed to work with old eigenvalue executioners such as 'NonlinearEigen'.  It is suggested to use the new eigenvalue executioner 'Eigenvalue' along with kernel tagging");
```

Reply:
I had to move the error (didn't think about the case of an EigenKernel with eigen=false in a non-EigenSystem), but I replaced the message with yours.